nio/util/logging/handlers/publisher/proxy.py

```diff
@@ -75,6 +76,12 @@ def publish(cls, signals):
         if cls._publisher_ready_event.is_set():
             with cls._publisher_lock:
                 cls._publisher.send(signals)
+        elif cls._open_thread and isinstance(cls._open_thread.nio_exception,
```

Comment:
Would we not want this to be the case for any exception that is raised in the thread? Should it only be just `PublisherNotReadyExceptions` for some reason?

I would propose this for the if statement instead:
```python
elif isinstance(getattr(cls._open_thread, 'nio_exception', None), Exception):
```